app/controllers/auth/JWTOpenIdAuthenticatedAction.scala

```diff
@@ -0,0 +1,109 @@
+package controllers.auth
+
+import com.jayway.jsonpath.JsonPath
+import net.minidev.json.JSONArray
+import pdi.jwt.{Jwt, JwtOptions, JwtAlgorithm, JwtClaim, JwtJson}
+import play.api.mvc._
+import pdi.jwt._
+import play.api.{Configuration, Logging}
+import scalaj.http.{Http, HttpOptions}
+import play.api.libs.json.Json
+import java.util.Base64
+import scala.util.Success
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class JWTOpenIdAuthenticatedAction(parser: BodyParsers.Default, appConfig: Configuration)(implicit ec: ExecutionContext)
+  extends ActionBuilderImpl(parser) with Logging {
+
+  logger.warn("In JWTOpenIdAuthenticatedAction")
+
+  private val JWT_LOGIN_URL = getValueFromConfigWithFallback("smui.JWTOpenIdAuthenticatedAction.login.url", "")
+  private val JWT_COOKIE = getValueFromConfigWithFallback("smui.JWTOpenIdAuthenticatedAction.cookie.name", "jwt")
+  private val JWT_AUTHORIZED_ROLES = getValueFromConfigWithFallback("smui.JWTOpenIdAuthenticatedAction.authorization.roles", "admin")
+
+  private lazy val authorizedRoles = JWT_AUTHORIZED_ROLES.replaceAll("\\s", "").split(",").toSeq
+
+  private def getValueFromConfigWithFallback(key: String, default: String): String = {
+    appConfig.getOptional[String](key) match {
+      case Some(value: String) => value
+      case None =>
+        logger.warn(s":: No value for $key found. Setting pass to super-default.")
+        default
+    }
+  }
+
+
+
+  def decodeRawAll(jwt: String): Try[(String, String, String)] = {
+    Jwt
+    .decodeRawAll(
+      jwt,
+      JwtOptions(signature = false, expiration = false, notBefore = false)
+    )
+  }
+
+  //private def isAuthenticated(jwt: String): Option[JwtClaim] = {
+  private def isAuthenticated(jwt: String): Option[Boolean] = {
```

Comment:
I am struggling mightly on the decodeAllRaw() method.  I get back the three results (header, claim, signature) but it's all wrapped in either a Try, a Future, or a Success, and I can't get to the actual claim itself!  I am looking to pull out the claim.scope to check if you are authenticated.

Reply:
This can be resolved through @mkr great work!